extra/mariadbfrm/mariadb_frm.cc

```diff
@@ -0,0 +1,498 @@
+/* Copyright (c) 2024, MariaDB
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1335 USA */
+
+/**
+  @file extra/mariadb_frm.cc
+  FRM file parser utility - extracts table structure from .frm files
+*/
+
+#include "mariadb.h"
+
+#include "my_sys.h"
+#include "mysql_com.h"
+#include "m_ctype.h"
+#include "field.h"
+#include "sql_class.h"
+#include "sql_show.h"
+#include "table_cache.h"
+#include "sql_table.h"
+
+#define DEBUG(fmt, ...)                                                       \
+  do                                                                          \
+  {                                                                           \
+    fprintf(stderr, "DEBUG: " fmt, ##__VA_ARGS__);                            \
+    fflush(stderr);                                                           \
+  } while (0)
+
+extern mysql_mutex_t LOCK_start_thread, LOCK_status,
+    LOCK_global_system_variables, LOCK_user_conn, LOCK_thread_id;
+extern mysql_cond_t COND_start_thread;
+extern struct system_variables global_system_variables;
+
+PSI_mutex_key key_LOCK_start_thread= 0;
+PSI_mutex_key key_LOCK_status= 0;
+PSI_mutex_key key_LOCK_global_system_variables= 0;
+PSI_mutex_key key_LOCK_user_conn= 0;
+PSI_mutex_key key_LOCK_thread_id= 0;
+PSI_cond_key key_COND_start_thread= 0;
+PSI_mutex_key key_LOCK_plugin= 0;
```

Comment:
do we really need these? I think they are already initialized to zeroes in mysqld.cc.

Reply:
`key_LOCK_thread_id` and `key_LOCK_plugin` are required, others I will remove.